### ISSUE TITLE
Support @CheckForNull annotation

### ISSUE THREAD

**Issue by ben-manes** [uber/NullAway#99]
```
error: [NullAway] method returns @Nullable, but superclass method com.github.benmanes.caffeine.cache.AsyncLoadingCache.getIfPresent(java.lang.Object) returns @NonNull
  public @Nullable CompletableFuture<V> getIfPresent(@Nonnull Object key) {
```

However, this method is annotated with `@CheckForNull` which should implicitly mean `@Nullable`.

**Comment by msridhar** [comment]
Which library provides the `@CheckForNull` annotation?

**Comment by ben-manes** [comment]
`javax.annotation` (jsr305 / findbugs)

**Comment by msridhar** [comment]
Cool, this should be easy to support

**Comment by chengniansun** [comment]
Just for your information, Guava is migrating off from jsr305. 

https://github.com/google/guava/issues/2960

**Comment by uhafner** [comment]
Are you ok with a pull request that adds support for a `@ChecksForNull` annotation? From the code it looks like that I just need to update https://github.com/uber/NullAway/blob/cc7869b6c89e388a6e0f8d0a6faba342f7827be4/nullaway/src/main/java/com/uber/nullaway/Nullness.java#L149 and provide a test. 

Or is there something conceptional wrong with this issue? Supporting `ChecksForNull` would help projects a lot that use NullAway and [SpotBugs](https://spotbugs.github.io)

**Comment by msridhar** [comment]
I would be open to a PR that just adds `@CheckForNull` support.  @lazaroclapp please comment if you see an issue with that.

**Comment by lazaroclapp** [comment]
I am fine with it. The idea is just to make `CheckForNull` another supported nullable annotation, right? As an exact alias for `@Nullable`, correct? Reading the previous discussion, we were a bit concerned about overhead of equality checking so many annotations if we just supported a custom list. But given that there seems to be interest for this particular annotation, I'd say go ahead.

We can open a separate task to optimize `isNullableAnnotation` and profile later.

**Comment by uhafner** [comment]
> I am fine with it. The idea is just to make `@CheckForNull` another supported nullable annotation, right? As an exact alias for `@Nullable`, correct? 

Yes, just a synonym for `@Nullable`. (The reason is that SpotBugs treats `@Nullable` differently: such references might be null but should *not be* flagged by a tool. In Spotbugs the semantics of `@CheckForNull`  is the same as your `@Nullable` annotation). Then teams can use SpotBugs and NullAway to check the code...
